core/src/main/java/fr/sncf/osrd/envelope/Envelope.java

```diff
@@ -201,6 +201,12 @@ public double getTotalTime() {
         return ((double) getTotalTimeMS()) / 1000;
     }
 
+    /** Returns the time between two positions of the envelope */
+    public double getTimeBetween(double beginPos, double endPos) {
+        var sliced = make(slice(beginPos, endPos));
```

Comment:
you don't need to  run slice() to get the time between two points, you can get the time at the end, the time at the beginning, and substract. slice makes a copy of the envelope, which is quite heavy just to get the time between two points

Reply:
True, that was quite a dumb implementation. I fixed it.